src/figuresClasses.ts

```diff
@@ -1,19 +1,86 @@
-export interface Figure {
-
+interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (
+      a <= 0
+      || b <= 0
+      || c <= 0
+      || !this.isTriangle(a, b, c)
+    ) {
+      throw new Error('Invalid triangle sides');
+    }
+  }
+
+  shape = 'triangle';
+
+  private isTriangle(
+    a: number,
+    b: number,
+    c: number,
+  ): boolean {
+    return (
+      this.a + b > c
+      && a + c > b
+      && b + c > a
+    );
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
 
+    return Math
+      .sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
```

Comment:
Probably you should limit decimal numbers after comma - just like with the circle